src/main/java/frc/robot/subsystems/CDSSubsystem.java

```diff
@@ -47,45 +53,140 @@
+      
+    }
+  }
 
+  public void CDSBeltToggle(boolean reverse) {
+    if(reverse){
       CDSBeltController.getSparkMax().set(-Constants.CDSBeltSpeed);
       CDSBeltController.setIdleMode(IdleMode.kBrake);
-      CDSBeltControllerDirection.setString("Reverse");
+      SmartDashboard.putString("CDS Belt Direction", "Reverse");
+      SmartDashboard.putNumber("CDS Belt Speed", -Constants.CDSBeltSpeed);
 
     } else {
-      CDSWheelControllerOne.getSparkMax().set(Constants.CDSWheelControllerSpeed);
-      CDSWheelControllerDirection.setString("Forward");
-
-      CDSBeltController.getSparkMax().set(-Constants.CDSBeltSpeed);
-      CDSBeltControllerDirection.setString("Forward");
+      CDSBeltController.getSparkMax().set(Constants.CDSBeltSpeed);
+      SmartDashboard.putString("CDS Belt Direction", "Forward");
+      SmartDashboard.putNumber("CDS Belt Speed", Constants.CDSBeltSpeed);
     }
   }
 
-  public int[] getSensorStatus() {
-    int frontStatus = colorSensorOne.getProximity() > 800 ? 1 : 0;
-    int backStatus = backBeamBreak.get() ? 1 : 0;
-    int[] beamBreakArray = {frontStatus, backStatus};
+  public double getBeltSpeed() {
+    return CDSBeltController.getSpeed();
+  }
+
+  public double getWheelSpeed() {
+    return CDSWheelControllerOne.getSpeed();
+  }
+
+  public void stopCDS() {
+    // stops all motors in the CDS
+    CDSWheelControllerOne.getSparkMax().set(0.0);
+    CDSBeltController.getSparkMax().set(0.0);
+    SmartDashboard.putNumber("CDS Wheel Speed", 0.0);
+    SmartDashboard.putNumber("CDS Belt Speed", 0.0);
+  }
+
+  public void stopCDSWheel() {
+    // Stops only the centering wheels
+    CDSWheelControllerOne.getSparkMax().set(0.0);
+    SmartDashboard.putNumber("CDS Wheel Speed", 0.0);
+  }
+  
+  public boolean[] getSensorStatus() {
+    int[] sensorStatuses = colorSensors.getProximities();
+
+    boolean backStatus = sensorStatuses[0] > Constants.backSensorActivation;
+    boolean middleStatus = sensorStatuses[1] > Constants.middleSensorActivation;
+    boolean frontStatus = sensorStatuses[2] > Constants.frontSensorActivation;
+    boolean[] beamBreakArray = {backStatus, middleStatus, frontStatus};
+
+    int ballCount = 0;
+    for (boolean status : beamBreakArray) {
+      if (status) {
+        ballCount++;
+      }
+    }
+
+    SmartDashboard.putNumber("Ball Count", ballCount);
+
     return beamBreakArray;
   }
 
+  public int getNextOpenSensor(boolean[] sensorSta) {
+
+    boolean[] sensorStatus = getSensorStatus();
+    
+    // Start at 1 to skip over the centering wheel status 
+    for (int i=0; i < sensorStatus.length-1; i++) {
+      if (sensorStatus[i]) {
+        return i;
+      } 
+    }
+    return -1;
+  }
+
+  public void periodic() {
+    // Color sensing
+    String ballColor = senseColor();
+    SmartDashboard.putString("Ball Color", ballColor);
+    SmartDashboard.putBoolean("Ball Color Match", ballColor == allianceColor);
+    boolean[] sensorStatus = getSensorStatus();
```

Comment:
Doesn't seem like we need this now that the main logic is in a command. Can at least get rid of the getSensorStatus() because it isn't used, and get rid of pushing the ball color value to the SmartDashboard because senseColor() will already do that